msprime/demography.py

```diff
@@ -850,10 +850,19 @@ def insert_extra_populations(self, tables):
                     md["name"] = population.name
                 if description_in_metadata:
                     md["description"] = population.description
-                # TODO we could also try to merge in the ``extra_metadata``
-                # depending on whether `additional_properties` is set,
-                # but it doesn't really seem worth it.
-                tables.populations.add_row(md)
+                for k in population.extra_metadata:
+                    if k in properties:
+                        md[k] = population.extra_metadata[k]
+                # if we've still got invalid metadata and None is allowed then
+                # fall back to None
+                try:
+                    tables.populations.add_row(metadata=md)
```

Comment:
Hmm, not sure what's motivating this. We should have valid metadata, shouldn't we? Is it that we're missing required properties? Silently storing *nothing* then seems like bad behaviour, wouldn't an informative error be better?

Reply:
What's motivating this is the SLiM example of populations, where `None` is valid metadata. Originally I put this in as
```
if 'type' in schema and 'null' in schema['type'] and (('required' in schema) and (len(set(schema['required']) - set(md.keys())) == 0)):
    md = None
```
... but the `try... except` seemed nicer.

Reply:
But, yes - an informative error would be better! I'll omit this bit. (There is no way to ask that the metadata be None, but perhaps that's OK...)